core/src/main/java/io/opentelemetry/android/features/diskbuffering/SignalFromDiskExporter.kt

```diff
@@ -65,11 +62,23 @@ class SignalFromDiskExporter
          */
         @WorkerThread
         @Throws(IOException::class)
-        fun exportBatchOfLogs(): Boolean =
-            logRecordFromDiskExporter?.exportStoredBatch(
-                exportTimeoutInMillis,
-                TimeUnit.MILLISECONDS,
-            ) ?: false
+        fun exportBatchOfLogs(): Boolean = export(logStorage, logExporter::export)
+
+        private fun <T> export(
+            storage: SignalStorage<T>,
+            exporter: (Collection<T>) -> CompletableResultCode,
+        ): Boolean {
+            var rc = false
+            val iter = storage.iterator()
+            while (iter.hasNext()) {
+                val result = exporter(iter.next()).join(exportTimeoutInMillis, TimeUnit.MILLISECONDS)
+                // TODO: What to do if a random export in the middle here just fails?
```

Comment:
Calling `next()` from the iterator will delete the current item, so if a network export fails, I'd say we should stop the loop and try again later. Most likely, it'd be due to network connectivity issues anyway, which should come and go often, so chances are it could be fixed by the next try.

Reply:
I'll create an issue to track this and we can pick that up in a follow-up PR. Thanks for the feedback. 👍🏻